### PR TITLE
use pointers to pass events and service checks to the aggregator

### PR DESCRIPTION
**Pull request by arbll**
### What does this PR do?

Use pointers to the event and service check struct to pass event and service checks to the aggregator.

### Motivation

Asked in a review https://github.com/DataDog/datadog-agent/pull/2950#discussion_r255449516. Created another PR to avoid polluting the first one.

### CODE REVIEW

File: pkg/aggregator/aggregator.go [hunk 1]
```diff
@@ -286,23 +286,23 @@ func (agg *BufferedAggregator) handleSenderSample(ss senderMetricSample) {
 }
 
 // addServiceCheck adds the service check to the slice of current service checks
-func (agg *BufferedAggregator) addServiceCheck(sc metrics.ServiceCheck) {
+func (agg *BufferedAggregator) addServiceCheck(sc *metrics.ServiceCheck) {
 	if sc.Ts == 0 {
 		sc.Ts = time.Now().Unix()
 	}
 	sc.Tags = deduplicateTags(sc.Tags)
```
Comment (remeh) [inline pkg/aggregator/aggregator.go]: Do you think it's fine to modify the source object?

File: pkg/aggregator/aggregator.go [hunk 2]
```diff
@@ -286,23 +286,23 @@ func (agg *BufferedAggregator) handleSenderSample(ss senderMetricSample) {
...
 
-	agg.serviceChecks = append(agg.serviceChecks, &sc)
+	agg.serviceChecks = append(agg.serviceChecks, sc)
 }
 
 // addEvent adds the event to the slice of current events
-func (agg *BufferedAggregator) addEvent(e metrics.Event) {
+func (agg *BufferedAggregator) addEvent(e *metrics.Event) {
 	if e.Ts == 0 {
```
Comment (remeh) [inline pkg/aggregator/aggregator.go]: Same here?

**Review by remeh — APPROVED** [review]
I've just added questions for a sanity check but otherwise, 👍 !

**Comment by arbll** [comment]
Moving to 6.11, this PR is a bit scary

**Comment by arbll** [comment]
We are not going to merge that as it's too risky. It will also become obsolete with the performance work we have planned in Q4.

**Pull request closed without merging**